updated docs with selectAllByDefault

kwongz opened the pull request:
### Description

<!---
  Describe the Pull Request here. Add any references and info to help reviewers understand your changes.
-->

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable

![image](https://github.com/evidence-dev/evidence/assets/69778837/44d890c3-dac4-45f2-a6a5-d19ad1cba3bf)

![image](https://github.com/evidence-dev/evidence/assets/69778837/810eaf0d-12f8-4311-8dc9-26f896a683fb)

ItsMeBrianD commented on sites/docs/pages/components/dropdown.md:
```diff
@@ -76,14 +77,14 @@ Selected: {inputs.category4.value}
     name=category4
     value=category_name
     title="Select a Category"
+    defaultValue="%"
```
Why are these additions needed?

kwongz replied:
Not needed, but my initial reaction to seeing the defaultValue section, is to see the use of it in the example.

Currently, we have other examples of how it can be used in other sections of dropdown, but the actual defaultValue example in the docs does not show the prop being used. 

I was just curious if this would be a more explicit example for new users.

OLD:
![image](https://github.com/evidence-dev/evidence/assets/69778837/220f0c09-8070-4d10-a766-24d96590c3da)

NEW:
![image](https://github.com/evidence-dev/evidence/assets/69778837/2c067209-439d-4fc6-9d42-4027c8590c38)

archiewood replied:
@kwongz this example using the `<DropdownOption value=%>` is a workaround from before we had the multiselect dropdown, which had the "select all" button and a corresponding property.

I would consider removing the slot from the example and just use the prop

kwongz commented:
I also added explicit targeting of "%" by applying the defaultValue prop. The previous example selected the "%" without using the defaultValue prop. 

But if we are trying to suggest to the user using the defaultValue to be applied automatically I can remove it.

OLD:
![image](https://github.com/evidence-dev/evidence/assets/69778837/220f0c09-8070-4d10-a766-24d96590c3da)

NEW:
![image](https://github.com/evidence-dev/evidence/assets/69778837/2c067209-439d-4fc6-9d42-4027c8590c38)

kwongz commented:
Removed sql filter query from unrelated multiselect sections

Removed in Multi-select:
![image](https://github.com/evidence-dev/evidence/assets/69778837/7a5a9cc2-96a9-4c24-88f6-3cbdad84add5)

Removed in Multiple defaultValues section
![image](https://github.com/evidence-dev/evidence/assets/69778837/d98937bb-f44a-4ac6-8b04-9486cfacd20f)

kwongz commented:
New Default Value section with changes @archiewood @ItsMeBrianD 

![image](https://github.com/evidence-dev/evidence/assets/69778837/cb4f674a-1646-48b0-941a-d535010fb83d)
